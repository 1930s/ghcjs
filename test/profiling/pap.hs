import GHC.Stack

f p1 p2 p3 = p1 + p2 + p3

forceIt f' r = do
    print r -- when this line is removed GHCJS compiled program produces same output
    print =<< whoCreated r
    print =<< whoCreated (f 1 2 3)
    print =<< whoCreated (f' 3)

main = do
    print =<< whoCreated f
    let f' = f 1 2
    print =<< whoCreated f'
    let r = f' 3
    forceIt f' r
