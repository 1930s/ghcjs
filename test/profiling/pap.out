["Main.CAF (<entire-module>)","Main.main (Main.hs:(11,1)-(16,16))"]
["Main.CAF (<entire-module>)","Main.main (Main.hs:(11,1)-(16,16))"]
6
["Main.CAF (<entire-module>)","Main.main (Main.hs:(11,1)-(16,16))"]
["Main.CAF (<entire-module>)","Main.main (Main.hs:(11,1)-(16,16))","Main.forceIt (Main.hs:(5,1)-(9,31))"]
["Main.CAF (<entire-module>)","Main.main (Main.hs:(11,1)-(16,16))","Main.forceIt (Main.hs:(5,1)-(9,31))"]
